thoth/package_extract/image.py

```diff
@@ -281,75 +286,88 @@ def _run_apt_cache_show(
-        to_ret |= _get_lib_dir_symbols(os.path.join(path, "lib64"))
-    elif os.path.exists(os.path.join(path, "usr/lib32")):
-        to_ret |= _get_lib_dir_symbols(os.path.join(path, "usr/lib32"))
-        to_ret |= _get_lib_dir_symbols(os.path.join(path, "lib32"))
-    elif os.path.exists(os.path.join(path, "usr/lib")):
-        to_ret |= _get_lib_dir_symbols(os.path.join(path, "usr/lib"))
-        to_ret |= _get_lib_dir_symbols(os.path.join(path, "lib"))
-    else:
-        raise FileNotFoundError("No usr libraries were found")
+def _get_lib_dir_symbols(result: dict, container_path: str, path: str) -> None:
+    """Get library symbols from a directory."""
+    for so_file_path in glob.glob(os.path.join(container_path, path, "*.so*")):
+        # We grep for '0 A' here because all exported symbols are outputted by nm like:
+        # 00000000 A GLIBC_1.x or:
+        # 0000000000000000 A GLIBC_1.x
+        command = f"nm -D {so_file_path!r} | grep '0 A'"
 
-    return to_ret
+        # Drop path to the extracted container in the output.
+        so_file_path = so_file_path[len(container_path):]
 
+        _LOGGER.debug("Gathering symbols from %r", so_file_path)
+        command_result = run_command(command, timeout=120, raise_on_error=False)
+        if command_result.return_code != 0:
+            _LOGGER.warning(
+                "Failed to obtain library symbols from %r; stderr: %s, stdout: %s",
+                so_file_path,
+                command_result.stderr,
+                command_result.stdout
+            )
+            continue
 
-def _ld_config_symbols(path: str):
-    to_ret = set()
-    with cwd(os.path.join(path, "etc/")), open("ld.so.conf", "r") as f:
-        for line in f.readlines():
-            paths = glob.glob(line.strip())
-            for p in paths:
-                with open(os.path.join(path, p), "r") as conf_file:
-                    for conf_path in conf_file.readlines():
-                        to_ret |= _get_lib_dir_symbols(conf_path.strip()[1:])
+        if so_file_path not in result:
+            result[so_file_path] = set()
 
-    return to_ret
+        for line in command_result.stdout.splitlines():
+            columns = line.split(' ')
+            print(columns)
+            if len(columns) > 2:
+                result[so_file_path].add(columns[2])
 
 
-# NOTE: Commented out func call below until LD_LIBR... is propagated properly
-def _ld_env_symbols(path: str):
-    to_ret = set()
-    ld_paths = os.environ.get("LD_LIBRARY_PATH")
+def _ld_config_symbols(result: dict, path: str) -> None:
+    """Gather library symbols based on ld.so.conf."""
+    _LOGGER.debug("Gathering symbols based on ld.so.conf file")
+    try:
+        with cwd(os.path.join(path, "etc/")), open("ld.so.conf", "r") as f:
+            for line in f.readlines():
+                if line.startswith("include "):
+                    line = line[len("include "):]
+
+                if not line:
+                    continue
+
```

Comment:
```suggestion
if not include:
    _get_lib_dir_symbols(result, path, os.path.join(path, line.strip())
    continue
for conf_path in glob.glob(line[len("include "):].strip()):
    _ld_config_symbols(result, path, conf_path)
```

Something like this?
Will require changing the function signature to something like _ld_config_symbols(result, path, config_file)

Reply:
Can be `_get_lib_dir_symbols` function reused (see my change). I think it accepts container path and the directory (which is parsed from `ld.so.conf`).

Reply:
The thing I want to change is explicitly stating /etc/ld.so.conf because /etc/ld.so.conf can point to other config files that will also have `include /config/file` as a line so I think it may be better to pass it as an argument and then also change: 
```with cwd(/etc):```
to:
```with cwd(os.path.split(conf_file)[0]):```

Reply:
> The thing I want to change is explicitly stating /etc/ld.so.conf because /etc/ld.so.conf can point to other config files that will also have `include /config/file`

Good point :+1: In that case, what about using a stack/queue which will keep track of files/directories stated in ld.so.conf file and all the linked files? In a push-pop manner we would traverse them and analyze them (if a file -> parse it as `ld.so.conf`, if a dir -> use `_get_lib_dir_symbolds` to retrieve symbols).

Reply:
Yeah, that sounds like a good idea, I think it probably makes the most sense to do a queue because that's likely how ldd processes

Reply:
Implemented in 5414780, PTAL.